problem_builder/tasks.py

```diff
@@ -6,40 +6,52 @@
 from celery.task import task
 from celery.utils.log import get_task_logger
 from instructor_task.models import ReportStore
+from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import UsageKey
 from xmodule.modulestore.django import modulestore
 
 from .mcq import MCQBlock, RatingBlock
+from problem_builder import AnswerBlock
 from .sub_api import sub_api
 
 logger = get_task_logger(__name__)
 
 
 @task()
-def export_data(source_block_id_str, user_id):
+def export_data(source_block_id_str, block_types, user_id, get_root=True):
     """
     Exports student answers to all MCQ questions to a CSV file.
     """
     start_timestamp = time.time()
 
     logger.debug("Beginning data export")
-
-    block_key = UsageKey.from_string(source_block_id_str)
-    src_block = modulestore().get_item(block_key)
-    course_key = src_block.scope_ids.usage_id.course_key.replace(branch=None, version_guid=None)
+    try:
+        block_key = UsageKey.from_string(source_block_id_str)
+        src_block = modulestore().get_item(block_key)
+        course_key = src_block.scope_ids.usage_id.course_key.replace(branch=None, version_guid=None)
+    except InvalidKeyError:
+        raise ValueError("Could not find the specified Block ID.")
     course_key_str = unicode(course_key)
 
-    # Get the root block:
     root = src_block
-    while root.parent:
-        root = root.get_parent()
+    if get_root:
+        # Get the root block for the course.
+        while root.parent:
+            root = root.get_parent()
+
+    type_map = {cls.__name__: cls for cls in [MCQBlock, RatingBlock, AnswerBlock]}
+
+    if not block_types:
+        block_types = tuple(type_map.values())
+    else:
+        block_types = tuple([type_map[class_name] for class_name in block_types])
```

Comment:
Nit: You can remove `[` and `]` so this is a generator expression (which is more efficient) rather than a list comprehension. Not significantly different here but it's a good habit.